File: go/leap/leap.go
// Package leap provides functionality on assessing whther a certain year is a
// leap year 
package leap

// IsLeapYear assesses whther a certain year is a leap year
func IsLeapYear(year int) bool {
	if year % 4 == 0 {
		if year % 100 == 0 && year % 400 == 0 {
			return true
		} else if year % 100 == 0 {
			return false
		} else {
			return true
		}
	}
	return false
}

File: cpp/leap/leap.cpp
#include "leap.h"

namespace leap {
    bool is_leap_year(int year) {
        // Return true if year is evenly divisible by 4, and,
        // year is evenly divisible with 400 and not evenly divisible with 100.
        if ((year % 4 == 0) && ((year % 400 == 0) || (year % 100 != 0)))
                return true;

        return false;
    }
}  // namespace leap
